### DISCUSSION TITLE
Writing a mesh file of a non-triangle mesh

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7984] [Q&A] [answered]
Hello,

I have a non-triangle mesh. When I run `CGAL::IO::write_OFF`, the mesh I get in the OFF file only has triangle faces (and is wrong). Is it normal? As far as I know, the OFF format is not restricted to triangles meshes, and there's nothing said about that in the doc. I'm wondering whether I did a mistake in my code.

**Comment by stla** [comment]
And is it possible to read a mesh file of a non-triangle mesh? Because I want to show you an issue with this mesh: there's a segfault when I try to triangulate it, but only on Windows, not on Linux (however same versions of R and GCC). This segfault didn't occur when I used CGAL < 5.6.

**Comment by sloriot** [comment]
do you have a code sample?

**Reply by stla** [reply, marked as answer]
Sorry, my mistake, the problem indeed is in my code: I have a function `epeck2epick` that I firstly called before writing the file (because there was a problem with EPECK) and this function only copies the first three indices of each face. I don't remember why I don't use `copy_face_graph` (maybe because this changes the order of the vertices or faces).
